components/reddit/reddit.app.js

```diff
@@ -224,18 +224,21 @@ module.exports = {
         typeahead_active: false,
       };
       const redditCommunities = await this._withRetries(() =>
-        this._makeRequest({
+         this._makeRequest({
           path: `/subreddits/search`,
           params,
         })
       );
+      return redditCommunities;
     },
     async getAllSearchSubredditsResults(query) {
       const results = [];
       let after = null;
       do {
         const redditCommunities = await this.searchSubreddits(after, query);
-        if (!redditCommunities) {
+        if ((redditCommunities && redditCommunities.data && !redditCommunities.data.children.length)
+          || !redditCommunities
+          || !redditCommunities.data) {
```

Comment:
You should use something like [`lodash/get`](https://lodash.com/docs/4.17.15#get) for this, and use a variable so that the code becomes cleaner/clearer:

```suggestion
        const isNewDataAvailable = get(redditCommunities, [
          "data",
          "children",
          "length",
        ]);
        if (!isNewDataAvailable) {
```